packages/form-js-viewer/src/render/hooks/useValuesAsync.js

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from 'preact/hooks';
+import useService from './useService';
+
+/**
+ * @enum { String }
+ */
+export const LOAD_STATES = {
+  LOADING: 'loading',
+  LOADED: 'loaded',
+  ERROR: 'error'
+};
+
+/**
+ * @typedef {Object} ValuesGetter
+ * @property {Object[]} values - The values data
+ * @property {(LOAD_STATES)} state - The values data's loading state, to use for conditional rendering
+ */
+
+/**
+ * A hook to load values for single and multiselect components.
+ *
+ * @param {Object} field - The form field to handle values for
+ * @return {ValuesGetter} valuesGetter - A values getter object providing loading state and values
+ */
+export default function(field) {
```

Comment:
I like this approach. Even though it might not be needed _right now_, this will help us when adding something like lazy data loading in the future. It also defines the data loading in a central place for all form fields 👍 Something we could have for conditional rendering as well.